src/main/java/ch/uzh/ifi/hase/soprafs23/SpooncularAPI/APIService.java

```diff
@@ -113,31 +125,41 @@ public RecipeDetailInfo getRecipeDetails(Long externalRecipeId) {
         }
     }
 
+    @PostConstruct
+    public void init() {
+        long count = fullIngredientRepository.count();
 
-    public List<String> getListOfIngredients(String initialString) { //for frontend to be displayed in order for the users to choose from
-        String query = URLEncoder.encode(initialString, StandardCharsets.UTF_8);
-        String ingredientsApiUrl = "https://api.spoonacular.com/food/ingredients/search?apiKey=" + apiKey + "&query=" + query + "&number=100";
+        // If the FullIngredient table is already populated, return and do not fetch ingredients from the API
+        if (count > 0) {
```

Comment:
wouldn't it be better to count to at least 100? 1000? in case there was sth wrong with fetching all ingredients from the API.

Another thought: shall we implement an error in case the API has an issue and its repeatingly gives too less ingredients (or is off in general) --> for the ingredient page we could say sth like "there was an error fetching all the provided ingredient names by the API. But you can still type them and send a recipe request" Should we discuss that wie frontend? Is the scenario of broken ingredient connection to the API, while at the same time being able to provide a recipe, reasonable? Or do we assume if its broken, its broken 100%?